crossplane-infra-access/README.md

```diff
@@ -0,0 +1,50 @@
+# Crossplane infrastructure access
+
+## Intro
+
+Team Honeybadger is deploying Crossplane to MCs. Crossplane can be used to manage infrastructure on the cloud provider side by both us and out customers.
+
+In order to do that, Crossplane has to be able to authenticate and authorize against the infrastructure provider. The access model that Crossplane uses depends on the user. By default, Crossplane doesn't require nor provide any access info. The administrator of Crossplane can create as many access secrets as he/she wants and can grant access to them to other Kubernetes users using the standard RBAC configuration.
+
+Example:
+1. Crossplane admin logins to an AWS account. In IAM, it creates two roles:
+  - `s3-access` that grants permissions to use S3 only,
+  - `db-access` that allows usage of RDS,
+2. For both roles, the admin creates access keys. Then, access keys are saved as secrets in the kubernetes API.
+3. The admin creates 2 `ProviderConfig` custom resources, named `s3` and `rds`, referencing respective Secrets with AWS API keys. Access to `ProviderConfig` is restricted.
+4. The admin grants permissions with RBAC:
+  - he/she allows access to `rds` `ProviderConfig` to the `db-automation` ServiceAccount in the `devel` Namespace
+  - he/she allows access to the `s3` `ProviderConfig` to the `customer-admins` Group
+5. From now on, respective SA and users belonging to the group can request Crossplane to manage (CRUD) resources in AWS.
+
+## Problem
+
+We want to configure all our MCs so that it's easy to use Crossplane by our internal (Giant Swarm) users.
+
+Example:
+1. We want to deploy Harbor to all CAPA MCs. Harbor needs an S3 bucket to work. How can we pre-configure MCs, so that it is possible to create the necessary buckets as part of Harbor App deployment?
+
+Specific questions in regard to infrastructure provider access tokens:
+
+1. Is it possible to easily mass-create a set of access credentials across multiple/all MCs?
```

Comment:
No, we need permissions to create IAM roles and policies. I would think more into a script or template that customer can use. For AWS, we have [TF script](https://github.com/giantswarm/giantswarm-aws-account-prerequisites) that helps with that

Reply:
We should extend our role creation scripts to add additional roles for the things we need access to through Crossplane